### PR TITLE
feat: add auth require setting for accessing user info

### PR DESCRIPTION
**Pull request by MayankMittal1**
Fixes: #581 
added use auth for user api setting in organization page
Screenshot from 2022-03-20 02-31-18

### CODE REVIEW

**Comment by hsluoyz** [comment]
@Steve0x2a plz review

**Comment by Steve0x2a** [comment]
@MayankMittal1 hi, /api/get-user and /api/get-users are two different apis, and https://github.com/casdoor/casdoor/issues/581 refers to /api/get-user.
Also, /api/get-users is protected by [authz](https://github.com/casdoor/casdoor/blob/master/authz/authz.go), I don't think /api/get-users needs another protection at the moment. Since only users of the built-in organization can use this api, and the built-in organization is seen as the global administrator, it seems unnecessary to control permissions for them.

I think it would be meaningful if it could be changed to protect /api/get-user, because currently /api/get-user only needs to know the organization name and username for anyone to get the user information.

**Comment by MayankMittal1** [comment]
@Steve0x2a okay, will do the necessary changes

**Comment by MayankMittal1** [comment]
@Steve0x2a merge plz

**Review by Steve0x2a — APPROVED** [review]
lgtm

**Comment by MayankMittal1** [comment]
@hsluoyz merge please

File: object/organization.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ type Organization struct {
 	Tags               []string `xorm:"mediumtext" json:"tags"`
 	MasterPassword     string   `xorm:"varchar(100)" json:"masterPassword"`
 	EnableSoftDeletion bool     `json:"enableSoftDeletion"`
+	UseAuthForUserApi  bool     `json:"useAuthForUserApi"`
```
Comment (hsluoyz) [inline object/organization.go:38]: Change to: `IsProfilePublic`

File: controllers/user.go [hunk 2]
```diff
@@ -87,6 +87,41 @@ func (c *ApiController) GetUser() {
...
+		hasPermission := false
+		if strings.HasPrefix(requestUserId, "app/") {
+			hasPermission = true
+		} else {
+			requestUser := object.GetUser(requestUserId)
+			if requestUser == nil {
+				c.ResponseError("Session outdated. Please login again.")
+				return
+			}
+			if requestUser.IsGlobalAdmin {
```
Comment (hsluoyz) [inline controllers/user.go:112]: We better use Casbin authorization to control permission here

File: controllers/user.go [hunk 3]
```diff
@@ -87,6 +87,41 @@ func (c *ApiController) GetUser() {
 	id := c.Input().Get("id")
 	owner := c.Input().Get("owner")
 	email := c.Input().Get("email")
+	organization := object.GetOrganization(fmt.Sprintf("%s/%s", "admin", strings.Split(id, "/")[0]))
```
Comment (hsluoyz) [inline controllers/user.go:90]: new line

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: controllers/user.go (hunk 2 above)
Comment (MayankMittal1) [reply controllers/user.go:112]: but we want to have this permission changable with every organisation, using casbin auth will require declaring at a global level, with no choice,
i have used similar approach as used in set-password api to verify permissions

**Pull request closed without merging**